cgen error for overriden `==` operator defined on a type alias

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 ebf629d, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 393d5b9af732cf6500437dd5e7fd62934722db02.ebf629d
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 20.04.6 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
|Memory              |1.42GB/15.05GB
|                    |
|V executable        |/home/delian/v/v
|V last modified time|2025-11-10 07:55:18
|                    |
|V home dir          |OK, value: /home/delian/v
|VMODULES            |OK, value: /home/delian/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/delian/v
|                    |
|env VFLAGS          |"-no-parallel"
|Git version         |git version 2.50.0
|V git status        |weekly.2025.46
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|gcc version         |gcc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|clang version       |Ubuntu clang version 18.1.8 (11~20.04.2)
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.31-0ubuntu9.18) 2.31

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  bb.v && bb`
```v
type MyInt = int
fn (i1 MyInt) == (i2 MyInt) bool { return int(i1) == int(i2) }
println( MyInt(3) == MyInt(1) )

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/bb.01K9QKAZEH831PVHWN8FXMWDA4.tmp.c:1438: warning: implicit declaration of function 'builtin__main__MyInt__eq'
cc: /home/delian/v/thirdparty/tcc/lib/libgc.a: error: 'GC_noop1_ptr' defined twice
cc: tcc: error: undefined symbol 'builtin__main__MyInt__eq'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```

**What did you expect to see?**

a compiled program printing false


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -164,7 +164,8 @@ fn (mut g Gen) infix_expr_eq_op(node ast.InfixExpr) {
 			g.styp(left.unaliased.set_nr_muls(0))
 		}
 		mut is_builtin_or_alias_to_builtin := left.sym.is_builtin()
-		if !is_builtin_or_alias_to_builtin && left.sym.info is ast.Alias {
+		if !has_alias_eq_op_overload && !is_builtin_or_alias_to_builtin
+			&& left.sym.info is ast.Alias {
 			alias_info := left.sym.info as ast.Alias
 			parent_sym := g.table.sym(alias_info.parent_type)
 			is_builtin_or_alias_to_builtin = parent_sym.is_builtin()

--- a/vlib/v/tests/aliases/alias_eq_op_test.v
+++ b/vlib/v/tests/aliases/alias_eq_op_test.v
@@ -1,0 +1,10 @@
+type MyInt = int
+
+fn (i1 MyInt) == (i2 MyInt) bool {
+	return int(i1) == int(i2)
+}
+
+fn test_main() {
+	c := MyInt(3) == MyInt(1)
+	assert c == false
+}

